src/software/geom/polynomial.cpp

```diff
@@ -0,0 +1,50 @@
+#include "software/geom/polynomial.h"
+
+Polynomial::Polynomial(const std::vector<double>& coeffs) : coeffs(coeffs)
+{
+    pruneCoeffs();
+}
+
+Polynomial::Polynomial(const std::initializer_list<double>& coeffs) : coeffs(coeffs)
+{
+    pruneCoeffs();
+}
+
+Polynomial::Polynomial(const std::pair<double, double>& p1,
+                       const std::pair<double, double>& p2)
+{
+    double slope = (p2.second - p1.second) / (p2.first - p1.first);
+    coeffs.push_back(p1.second - (p1.first * slope));
+    coeffs.push_back(slope);
+}
+
+const std::vector<double>& Polynomial::getCoeffs() const
+{
+    return coeffs;
+}
+
+double Polynomial::calculateValue(double val) const
```

Comment:
I feel like a function similar to 'valueAt(double val)' would explain the function clearer

Reply:
done